symfony2/yaml/forms.yml

```diff
@@ -53,4 +53,58 @@ questions:
             - {value: "validation_groups", correct: true}
             - {value: "groups_validation", correct: false}
             - {value: "validator_groups",  correct: false}
-            - {value: "groups_validator",  correct: false}
\ No newline at end of file
+            - {value: "groups_validator",  correct: false}
+    -
+        question: 'Which form event exist?'
+        answers:
+            - {value: "FormEvents::PRE_SET_DATA",   correct: true}
+            - {value: "FormEvents::SUBMIT",         correct: true}
+            - {value: "FormEvents::POST_SET_DATA",  correct: true}
+            - {value: "FormEvents::POST_REQUEST",   correct: false}
+    -
+        question: 'What data is inside FormEvent object at FormEvents::PRE_SET_DATA?'
+        answers:
+            - {value: "Model data",             correct: true}
+            - {value: "Model data normalized",  correct: false}
+            - {value: "Request data",           correct: false}
+            - {value: "View data",              correct: false}
+    -
+        question: 'What data is inside FormEvent object at FormEvents::PRE_SUBMIT?'
+        answers:
+            - {value: "Model data",             correct: false}
+            - {value: "Model data normalized",  correct: false}
+            - {value: "Request data",           correct: true}
+            - {value: "View data",              correct: false}
+    -
+        question: 'Which text type exist?'
+        answers:
+            - {value: "money",    correct: true}
+            - {value: "currency", correct: true}
+            - {value: "textarea", correct: true}
+            - {value: "surname",  correct: false}
+    -
+        question: 'Which date type exist?'
+        answers:
+            - {value: "date",       correct: true}
+            - {value: "datetime",   correct: true}
+            - {value: "time",       correct: true}
+            - {value: "timestamp",  correct: false}
+    -
+        question: 'How do you attach a constraint to a form field?'
```

Comment:
This question is too ambigous, because you do not attach contraint to the form field but you bind a constraint to a property of your model. It's hidden because form component hug the validator and do it internally for you.

So answer `Add an annotation in model` is true via `@assert` annotation.

You can look at validator component : http://symfony.com/fr/doc/current/book/validation.html, you also can create a mapping validation (yml, xml) to achieve that.